pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -669,11 +669,23 @@ public void close() throws IOException {
         });
 
         if (listenChannel != null) {
-            listenChannel.close();
+            try {
+                if (!listenChannel.close().await(30, TimeUnit.SECONDS)) {
```

Comment:
Please make timeout configurable.

Reply:
Thanks for the review @sijie .
I'll move this change to a separate PR as you have suggested for one of the other changes.

Reply:
I have removed this change from this PR. The new PR for closing the channels is #9308 . PTAL @sijie